agents/research_agent.py

```
@@ -244,24 +273,83 @@ def _parse_text_result(self, text_result: str, query: str) -> Dict[str, Any]:
             else:
-                parsed_result = result
-                
+                parsed_result = result if isinstance(result, dict) else {'raw_result': str(result)}
+
+            # Ensure required fields exist
+            if 'part_number' not in parsed_result and 'query' not in parsed_result:
+                parsed_result['part_number'] = query
+
+            return parsed_result
+
+        except Exception as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)